morpheus/_lib/src/messages/multi_inference_fil.cpp

```diff
@@ -73,7 +73,7 @@ std::shared_ptr<MultiInferenceFILMessage> MultiInferenceFILMessageInterfaceProxy
 std::shared_ptr<morpheus::InferenceMemory> MultiInferenceFILMessageInterfaceProxy::memory(
     MultiInferenceFILMessage &self)
 {
-    return self.memory;
+    return DCHECK_NOTNULL(std::dynamic_pointer_cast<morpheus::InferenceMemory>(self.memory));
```

Comment:
Can this be a `std::static_pointer_cast` instead? We can add a `DCHECK` beforehand that does the dynamic cast just to be sure.